go/src/koding/kites/kloud/provider/azure/stack.go

```diff
@@ -99,6 +101,14 @@ func (s *Stack) VerifyCredential(c *stack.Credential) error {
 	return nil
 }
 
+func trim(s string, n int) string {
```

Comment:
It's not a single allocation, it's three allocations - it allocates `*bytes.Buffer`, `[]byte` from string and string from `[]byte`, unnecessarily.

We'd need to live with the function, I can eventually inline it. We need to make the code better allocation-wise, not worse, even though it may be not perfect now.